### ISSUE TITLE
What should ````workspace.lexos```` consist of

### ISSUE THREAD

**Issue by chantisnake** [WheatonCS/Lexos#197]
There is two possibility:

#### current code: 
1. consist of the whole session folder, and the ````session.p````.
2. this is handled by put the ziped session folder in to current session folder, and then unpickle ````session.p```` and delete ````session.p````
3. this method will keep all the information of the file you previously work with, but if you upload anything before upload ````workspace.lexos````, all the information will be erased, because we cannot have 2 file manager, and we cannot merge 2 file manager

---
#### another version:
1. consist of files and the ````session.p````
2. we treat all the file as upload file and make in the upload process detect ````sesssion.p```` and unpickle and delete that.
3. this method is more consistent, all we need is a zip file handle and a ````session.p```` handle. but this will loss crucial information and cause error.
for example in multicloud, the upload file name is cached in session, so if the workspace is restored, there is no file uploaded but the js will display a file name there, which seems like that there is a file uploaded. 
I can prevent the file name showing up in this case, but for future development this loss of information is not good, and in the future there might need more and more complicated work to make this work.

----
I would like to get everyone's opinion on this. Thanks

**Comment by chantisnake** [comment]
the solution of the first one is to add a ````merge_filemanager()```` function in the general function. but that should be also complicated in the future

**Comment by mleblanc321** [comment]
we need to discuss this on Monday ...

**Comment by scottkleinman** [comment]
Yes, this will take some thought. I will meditate on it...

**Comment by mleblanc321** [comment]
don't mediate so long in dunkinDonuts that you miss your train :)

**Comment by scottkleinman** [comment]
There was little danger of that. The train was late and then further delayed by a police situation (is there a Red Sox game today?).

In any event, my initial feeling is that we should keep things simple, especially since this is our first stab at working with zip archives and saving sessions. We just have to detect a session.p file in the uploaded zip archive and then warn the user that continuing will erase their current session (*including their scrubbed and/or cut files*--so they have a chance to download them). If the user uploads a zip archive without a session.p file, Lexos will extract the document files in the archive and merge them into the current session. The user will then have to select those files and apply scrubbing and cutting tools again if they want the files to have the same features. If the files were previously downloaded after these tools had been applied, this is obviously not necessary.

**Comment by chantisnake** [comment]
workspace is stored in a file with .lexos extension, so we don't even need to check session.p

**Comment by mleblanc321** [comment]
i agree with scott:  let's implement (and feel confident about) small increments of functionality:
(1) upload .zip (containing no session, files only);  these should be merged into the current session
(but current session options cannot be associated with the new (merged) text files)
(2a) save session.p and include with files to download
(2b) upload .zip (containing session.p) must generate a WARNING dialog saying that "the uploaded file contains a previous session; continuing will erase the current session; do you want to continue? Y/N"

RE: (2) i'd like to see some metrics on just how big this file is

in short, the exercise forced Moses to *really* work hard to visit all the code where options are stored ...
